lib/lobby.ts

```diff
@@ -0,0 +1,157 @@
+import { getClient } from '.';
+import { DbItem, IDbItem } from './db-item';
+import { COLLECTION } from './private/enums';
+import { User } from './user';
+
+type DatabaseEntry = Omit<ILobby, 'collectionName' | 'existsInDb'>;
+type ClientResponse = Omit<DatabaseEntry, 'users' | 'uri'>;
+
+export interface LobbyProps {
+  /**
+  * The spotify id of playlist
+  */
+  readonly spotifyPlaylistId?: string;
+
+  /**
+   * The manager of a lobby
+   */
+  readonly manager: User;
+
+  /**
+   * The participants in a lobby
+   */
+  readonly participants: string[];
+
+  /**
+   * The theme of a lobby
+   */
+  readonly theme: string;
+
+  /**
+   * The playlist name
+   */
+  readonly name: string;
+}
+
+export interface ILobby extends Omit<LobbyProps, 'topSongs'>, IDbItem {
+}
+
+/**
+ * The class containing a user and their data
+ */
+export class Lobby extends DbItem implements ILobby {
+  /**
+   * A static function to query for a user from their id
+   *
+   * @returns a user object if the id exists in the database
+   */
+  public static async fromId(id: string): Promise<Lobby | null> {
+    const client = await getClient();
+    const document = await client.findDbItem(COLLECTION.LOBBIES, id);
+    if (!document) return null;
+    const content: DatabaseEntry = document.getContent() as DatabaseEntry;
+    return new Lobby(id, content, document.key ?? null);
```

Comment:
i dont think this works because User wont be passed in correctly!

Reply:
oh yea, do you think it would be better to change User to just be the userid?

Reply:
I think for the manager it’s acceptable to use the user because u will have to perform a write anyways

but since this write to the user only really happens whenever a user joins the lobby, leaves, or the lobby is deleted I think the id is sufficient